### ISSUE TITLE
11.1 more server crashes

### ISSUE THREAD

**Issue by padmanek** [StrangeLoopGames/EcoIssues#25218]
It seems like data raceived from the player failed to process which ended up crashing the server right after.
```
[22:03:14.449] [14] [Error] [Eco] Got exception at processing message from sphiix:                         [Ping: 88 ms]   [CPU: 0.3 ms]   [In: ∞ k/s]     [Out: ∞ k/s]    . Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Shared.Serialization.BSONValueFactory`1.New(BSONValueType valueType)
   at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
   at Eco.Shared.Serialization.SimpleBSON.DecodeObject(MemoryStream stream, BSONObject obj)
   at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
   at Eco.Shared.Serialization.SimpleBSON.DecodeArray(MemoryStream stream)
   at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
   at Eco.Shared.Serialization.SimpleBSON.DecodeObject(MemoryStream stream, BSONObject obj)
   at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
   at Eco.Shared.Networking.NetMessage.DecodePacket(ReadOnlySpan`1 message)
   at Eco.Plugins.Networking.NetworkServer.HandleIncomingDataMessage(UdpDataMessage& message). The message length is: 125 and text representation is objssendChunkssendObjectspos��DY֘B�єDtime��Erot>�="4w��tD��0�v���@�����y ?yaw8<�anim id�� 
[22:03:14.454] [21] [Error] [Eco] System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Shared.Utils.ByteColor.ToBson()
   at Eco.Shared.Serialization.BsonManipulator.ToBson(Object val, BsonSerializationContext& context, Boolean useReflection)
   at Eco.Shared.Utils.EnumerableExtensions.ToBson(IEnumerable enumeration, BsonSerializationContext& context, Boolean doAllParams)
   at Eco.World.WorldChunk.PackIntoBson()
   at Eco.World.PersistentChunk.PackIntoBson()
   at Eco.World.WorldChunkGrid.GetChunkUpdate(WorldChunk chunk)
   at Eco.World.Internal.ChunkUpdater.SyncQueues[T](Dictionary`2 queues)
   at Eco.World.Internal.ChunkUpdater.Service()
   at Eco.World.WorldPlugin.DoChunkUpdates()
   at Eco.Core.Utils.Threading.ActionUtils.<>c__DisplayClass1_0.<AsAsyncCancellableAction>b__0(CancellationToken _)
   at Eco.Core.Utils.Threading.IntervalActionWorker.DoRepeatableWork(CancellationToken token)
   at Eco.Core.Utils.Threading.RepeatableWorkerBase.DoWork(CancellationToken token)
   --- End of inner exception stack trace --- System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Shared.Utils.ByteColor.ToBson()
   at Eco.Shared.Serialization.BsonManipulator.ToBson(Object val, BsonSerializationContext& context, Boolean useReflection)
   at Eco.Shared.Utils.EnumerableExtensions.ToBson(IEnumerable enumeration, BsonSerializationContext& context, Boolean doAllParams)
   at Eco.World.WorldChunk.PackIntoBson()
   at Eco.World.PersistentChunk.PackIntoBson()
   at Eco.World.WorldChunkGrid.GetChunkUpdate(WorldChunk chunk)
   at Eco.World.Internal.ChunkUpdater.SyncQueues[T](Dictionary`2 queues)
   at Eco.World.Internal.ChunkUpdater.Service()
   at Eco.World.WorldPlugin.DoChunkUpdates()
   at Eco.Core.Utils.Threading.ActionUtils.<>c__DisplayClass1_0.<AsAsyncCancellableAction>b__0(CancellationToken _)
   at Eco.Core.Utils.Threading.IntervalActionWorker.DoRepeatableWork(CancellationToken token)
   at Eco.Core.Utils.Threading.RepeatableWorkerBase.DoWork(CancellationToken token)

```

**Comment by padmanek** [comment]
> That doesn't look like a crash, I can see such errors in the logs of the official servers rarely, but they don't lead to a crash.
> 
> Is there any more information?


The server runs on Ubuntu so I just got this from journalctl, it's from around the crash. That's all I have.
```
Oct 27 22:01:14 ubuntu EcoServer[653665]: [10:01:14] Removed backup - /root/gameservers/silverleaf/Storage/Backup/Game-2024-10-26-22.01.06
Oct 27 22:01:45 ubuntu EcoServer[653665]: [10:01:45] [NidToolbox]: Xquise has logged in with IP 102.119.236.5
Oct 27 22:01:47 ubuntu EcoServer[653665]: [10:01:47] [NidToolbox]: InvisibleInk777 has logged out.
Oct 27 22:01:57 ubuntu EcoServer[653665]: [10:01:57] [NidToolbox]: GIANGIELLA has logged in with IP 80.183.21.113
Oct 27 22:01:57 ubuntu EcoServer[653665]: [10:01:57] [NidToolbox]: ChOJO has logged out.
Oct 27 22:02:17 ubuntu EcoServer[653665]: [10:02:08] Sending 26205 minimap objects to Steam:76561198075697085:        [Ping: 111 ms]  [CPU: 2.6 ms]   [In: ∞ k/s]     [Out: ∞ k/s]    [10:02:13] Sending 26204 minimap objects to Steam:76561198178051878:        [Ping: 276 ms]  [CPU: 0.2 ms]   [In: ∞ k/s]     [Out: ∞ k/s]    [10:02:17] [NidToolbox]: InvisibleInk777 has logged in with IP 24.96.109.180
Oct 27 22:03:14 ubuntu EcoServer[653665]: [10:02:27] Sending 26202 minimap objects to InvisibleInk777:                [Ping: 384 ms]  [CPU: 2.0 ms]   [In: ∞ k/s]     [Out: ∞ k/s]    [10:03:14] System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
Oct 27 22:03:14 ubuntu EcoServer[653665]:   Inner Exception:
Oct 27 22:03:14 ubuntu EcoServer[653665]:   System.NullReferenceException: Object reference not set to an instance of an object.
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Shared.Utils.ByteColor.ToBson()
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Shared.Serialization.BsonManipulator.ToBson(Object val, BsonSerializationContext& context, Boolean useReflection)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Shared.Utils.EnumerableExtensions.ToBson(IEnumerable enumeration, BsonSerializationContext& context, Boolean doAllParams)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.WorldChunk.PackIntoBson()
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.PersistentChunk.PackIntoBson()
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.WorldChunkGrid.GetChunkUpdate(WorldChunk chunk)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.Internal.ChunkUpdater.SyncQueues[T](Dictionary`2 queues)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.Internal.ChunkUpdater.Service()
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.World.WorldPlugin.DoChunkUpdates()
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Core.Utils.Threading.ActionUtils.<>c__DisplayClass1_0.<AsAsyncCancellableAction>b__0(CancellationToken _)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Core.Utils.Threading.IntervalActionWorker.DoRepeatableWork(CancellationToken token)
Oct 27 22:03:14 ubuntu EcoServer[653665]:      at Eco.Core.Utils.Threading.RepeatableWorkerBase.DoWork(CancellationToken token)
Oct 27 22:03:14 ubuntu EcoServer[653665]:   
Oct 27 22:03:14 ubuntu EcoServer[653665]: [10:03:14] Got exception at processing message from sphiix:                         [Ping: 88 ms]   [CPU: 0.3 ms]   [In: ∞ k/s]     [Out: ∞ k/s]    . Exception: System.NullReferenceException: Object reference not set to an instance of an object.
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.BSONValueFactory`1.New(BSONValueType valueType)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeObject(MemoryStream stream, BSONObject obj)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeArray(MemoryStream stream)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeObject(MemoryStream stream, BSONObject obj)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Serialization.SimpleBSON.DecodeElement(MemoryStream stream)
Oct 27 22:03:14 ubuntu EcoServer[653665]:    at Eco.Shared.Networking.NetMessage.DecodePacket(ReadOnlySpan`1 message)
Oct 27 22:03:14 ubuntu EcoServer[653665]: [287B blob data]
Oct 27 22:03:14 ubuntu EcoServer[653665]: [18B blob data]
Oct 27 22:03:15 ubuntu EcoServer[653665]: [11B blob data]
Oct 27 22:03:15 ubuntu systemd[1]: EcoServer1.service: Main process exited, code=exited, status=1/FAILURE
Oct 27 22:03:15 ubuntu systemd[1]: EcoServer1.service: Failed with result 'exit-code'.
Oct 27 22:03:15 ubuntu systemd[1]: EcoServer1.service: Consumed 9h 8min 11.566s CPU time.
Oct 27 22:03:20 ubuntu systemd[1]: EcoServer1.service: Scheduled restart job, restart counter is at 2.
Oct 27 22:03:20 ubuntu systemd[1]: Stopped Silverleaf Eco 1.
Oct 27 22:03:20 ubuntu systemd[1]: EcoServer1.service: Consumed 9h 8min 11.566s CPU time.
Oct 27 22:03:20 ubuntu systemd[1]: Started Silverleaf Eco 1.
Oct 27 22:03:22 ubuntu EcoServer[656231]: [10:03:22] Logged in to Strange Cloud using username and password.
Oct 27 22:03:22 ubuntu EcoServer[656231]: [10:03:22] Eco Server 0.11.1.0 beta release-750, Logged in as padman
```

**Comment by padmanek** [comment]
It very much looks like processing data from a logging in player crashed the server.

**Comment by D3nnis3n** [comment]
Is there a chance you can say which user causes this, so we get any hint at what's wrong for reproduction. Maybe can even direct the user here?

**Comment by padmanek** [comment]
From the logs it looks like its the user is called `sphiix` since the server crashes the second he joins.

You can see on the first server log the message from the user says something about sending chunks and sending objects:

`at Eco.Plugins.Networking.NetworkServer.HandleIncomingDataMessage(UdpDataMessage& message). The message length is: 125 and text representation is ����objs�����sendChunks���sendObjects���pos����DY֘B�єD�time����E�rot�>��="4w��tD���0��v����@�����y ?�yaw�8<���anim� �id����`

And the error right after is also about processing the chunks `at Eco.World.WorldChunk.PackIntoBson()`

It's as if it was trying to serialize the gibberish it just got from that user joining and fails.

I can try to get you in touch with the player but he's totally unaware of this and I don't know if he is willing to help.

**Comment by D3nnis3n** [comment]
Can send the user id of that user to support@strangeloopgames.com? Maybe there is something off with their account.

**Comment by D3nnis3n** [comment]
ID was received, thanks!

**Comment by D3nnis3n** [comment]
A developer requested your savegame, can you provide that?

**Comment by padmanek** [comment]
https://drive.google.com/file/d/1loumOD3v_Q6-U1KweKPavGbid4JQe3ri/view?usp=sharing

Altho I am pretty sure all the crashes we've had are related to a **hardware issue** that I've discovered on our box.
I've moved our server to another machine last Thursday and we went from 3-4 crashes per day to 0 crashes since the migration.

So very very likely we can just close this ticket to not waste any more time on it.
